### ISSUE TITLE
play media issues

### ISSUE THREAD

**Issue by juha-h** [sipwise/rtpengine#780]
I tried to play a file
```
$ file /tmp/file.wav 
/tmp/file.wav: RIFF (little-endian) data, WAVE audio, Microsoft PCM, 16 bit, mono 8000 Hz
```
using 'play media' command.  Audio length of the file is about 12 seconds, but based on rtpengine debug output and called party debug output, 'play media' plays it in about 6 seconds.  Also, file played only to called party.

Below is start of the debug.  I can provide whole debug is needed.
```
May 23 10:44:21 salmon rtpengine[21362]: INFO: [1aeaab233429485e]: Received command 'play media' from 127.0.0.1:33152
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: Dump for 'play media' from 127.0.0.1:33152: { "supports": [ "load limit" ], "file": "/tmp/file.wav", "call-id": "1aeaab233429485e", "received-from": [ "IP4", "192.168.43.99" ], "from-tag": "15dbd9bee3f645d0", "to-tag": "4abe831eb0b87b2f", "command": "play media" }
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: creating media_player
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: shutting down media_player
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: av_log: Opening '/tmp/file.wav' for reading
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: av_log: Setting default whitelist 'file,crypto'
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: av_log: Probing wav score:99 size:2048
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: av_log: Format wav probed with size=2048 and score=99
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: av_log: Before avformat_find_stream_info() pos: 44 bytes read:65580 seeks:1 nb_streams:1
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: av_log: probing stream 0 pp:32
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: av_log: Probing mp3 score:1 size:4096
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: av_log: Probe with size=4096, packets=2469 detected mp3 with score=1
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: av_log: probing stream 0 pp:31
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: av_log: Probing mp3 score:1 size:8192
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: av_log: Probe with size=8192, packets=2470 detected mp3 with score=1
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: av_log: probing stream 0 pp:30
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: av_log: probing stream 0 pp:29
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: av_log: Probing mp3 score:1 size:16384
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: av_log: Probe with size=16384, packets=2472 detected mp3 with score=1
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: av_log: probing stream 0 pp:28
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: av_log: probing stream 0 pp:27
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: av_log: probing stream 0 pp:26
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: av_log: probing stream 0 pp:25
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: av_log: probing stream 0 pp:24
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: av_log: probing stream 0 pp:23
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: av_log: probing stream 0 pp:22
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: av_log: probing stream 0 pp:21
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: av_log: probing stream 0 pp:20
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: av_log: probing stream 0 pp:19
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: av_log: probing stream 0 pp:18
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: av_log: probing stream 0 pp:17
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: av_log: probing stream 0 pp:16
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: av_log: probing stream 0 pp:15
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: av_log: probing stream 0 pp:14
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: av_log: probing stream 0 pp:13
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: av_log: probing stream 0 pp:12
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: av_log: probing stream 0 pp:11
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: av_log: probing stream 0 pp:10
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: av_log: probing stream 0 pp:9
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: av_log: probed stream 0
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: av_log: parser not found for codec pcm_s16le, packets or times may be invalid.
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: av_log: max_analyze_duration 5000000 reached at 5120000 microseconds st:0
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: av_log: stream 0: start_time: -1152921504606847.000 duration: 5.784
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: av_log: format: start_time: -9223372036854.775 duration: 5.784 bitrate=128 kb/s
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: av_log: After avformat_find_stream_info() pos: 92588 bytes read:158124 seeks:1 frames:22
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: Output codec for media playback is PCMU/8000
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: Creating SSRC transcoder from PCM-S16LE/8000/1 to PCMU/8000/1
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: Initialized encoder with frame size 160 samples
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: Encoder created with clockrate 8000, 1 channels, using sample format 1 (ptime 20 for 160 samples per frame and 160 samples (160 bytes) per packet, bitrate 0)
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: Created media playback context for PCM-S16LE/8000 -> PCMU/8000
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: read media packet: pts 0 duration 2048 (scaled 0/2048, 256000 us), sample rate 8000, time_base 1/8000
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: RTP media successfully decoded: TS 0, samples 2048
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: RTP media successfully encoded: TS 0, len 160
May 23 10:44:21 salmon /usr/bin/sip-proxy[10926]: INFO: Routing in-dialog ACK <sip:jh-0x55a56674a2d0@192.168.43.99:35795;transport=tcp> from <sip:test@test.tutpro.com> to <sip:192.168.43.99:40466;transport=tcp> based on gruu
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: Adding 160 bytes to packetizer
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: Received packet of 160 bytes from packetizer
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: Scheduling to send RTP packet (seq 23932 TS 1095918592) at 1558597461.182028
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: RTP media successfully encoded: TS 160, len 160
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: Adding 160 bytes to packetizer
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: Received packet of 160 bytes from packetizer
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: Scheduling to send RTP packet (seq 23933 TS 1095918752) at 1558597461.202028
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: RTP media successfully encoded: TS 320, len 160
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: Adding 160 bytes to packetizer
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: Received packet of 160 bytes from packetizer
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: Scheduling to send RTP packet (seq 23934 TS 1095918912) at 1558597461.222028
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: RTP media successfully encoded: TS 480, len 160
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: Adding 160 bytes to packetizer
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: Received packet of 160 bytes from packetizer
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: Scheduling to send RTP packet (seq 23935 TS 1095919072) at 1558597461.242028
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: RTP media successfully encoded: TS 640, len 160
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: Adding 160 bytes to packetizer
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: Received packet of 160 bytes from packetizer
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: Scheduling to send RTP packet (seq 23936 TS 1095919232) at 1558597461.262028
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: RTP media successfully encoded: TS 800, len 160
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: Adding 160 bytes to packetizer
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: Received packet of 160 bytes from packetizer
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: Scheduling to send RTP packet (seq 23937 TS 1095919392) at 1558597461.282028
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: RTP media successfully encoded: TS 960, len 160
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: Adding 160 bytes to packetizer
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: Received packet of 160 bytes from packetizer
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: Scheduling to send RTP packet (seq 23938 TS 1095919552) at 1558597461.302028
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: RTP media successfully encoded: TS 1120, len 160
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: Adding 160 bytes to packetizer
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: Received packet of 160 bytes from packetizer
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: Scheduling to send RTP packet (seq 23939 TS 1095919712) at 1558597461.322028
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: RTP media successfully encoded: TS 1280, len 160
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: Adding 160 bytes to packetizer
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: Received packet of 160 bytes from packetizer
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: Scheduling to send RTP packet (seq 23940 TS 1095919872) at 1558597461.342028
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: RTP media successfully encoded: TS 1440, len 160
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: Adding 160 bytes to packetizer
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: Received packet of 160 bytes from packetizer
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: Scheduling to send RTP packet (seq 23941 TS 1095920032) at 1558597461.362028
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: RTP media successfully encoded: TS 1600, len 160
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: Adding 160 bytes to packetizer
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: Received packet of 160 bytes from packetizer
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: Scheduling to send RTP packet (seq 23942 TS 1095920192) at 1558597461.382028
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: RTP media successfully encoded: TS 1760, len 160
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: Adding 160 bytes to packetizer
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: Received packet of 160 bytes from packetizer
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: Scheduling to send RTP packet (seq 23943 TS 1095920352) at 1558597461.402028
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: Forward to sink endpoint: 192.168.43.99:8714 (RTP seq 23932 TS 1095918592)
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: Forward to sink endpoint: 192.168.43.99:8714 (RTP seq 23933 TS 1095918752)
May 23 10:44:21 salmon rtpengine[21362]: DEBUG: [1aeaab233429485e]: queuing up packet for delivery at 1558597461.222028 (RTP seq 23934 TS 1095918912)
```

**Comment by rfuchs** [comment]
Can you share that file? The length is based on what's reported by ffmpeg.

And yes, media is played only to one party, the one that is selected via SIP tags, label, or address.

**Comment by juha-h** [comment]
Sorry,my mistake. I had used different file than I though.  The actual file length was 6 seconds. And I got the file played to both parties of the call by calling play_media() twice with different tags:
```
play_media("file=/tmp/file.wav to-tag=$tt");
play_media("file=/tmp/file.wav from-tag=$ft");
```
One comment.  Now the played media is mixed to the end-to-end stream.  Have you though about possibility to choose that end-to-end stream is interrupted while the media is being played?
